src/subscription_manager/managercli.py

```diff
@@ -309,6 +309,38 @@ def __init__(self, name="cli", shortdesc=None, primary=False):
     def _get_logger(self):
         return logging.getLogger('rhsm-app.%s.%s' % (self.__module__, self.__class__.__name__))
 
+    def test_connection(self, proxy_host, proxy_port, proxy_user, proxy_password):
```

Comment:
I think Cli will have already created a consumer_cp and a no_auth_cp with the info form cli/config by the time test_connection is called  (if self.require_connection() is true...).

Could self.no_auth_cp be used for test_connection?